frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -768,6 +781,365 @@ class SegmentsView extends React.Component<Props, State> {
+        </React.Fragment>,
+      );
+    }
+  };
+
+  getSetGroupColorMenuItem = (id: number): ItemType => {
+    const isEditingDisabled = !this.props.allowUpdate;
+    return {
+      key: "changeGroupColor",
+      disabled: isEditingDisabled,
+      label: (
+        <ChangeColorMenuItemContent
+          title="Change Group Color"
+          isDisabled={isEditingDisabled}
+          onSetColor={(color) => {
+            if (getVisibleSegmentationLayer == null) {
+              return;
+            }
+            this.setGroupColor(id, color);
+          }}
+          rgb={[0.5, 0.5, 0.5]}
+        />
+      ),
+    };
+  };
+
+  getLoadMeshesMenuItem = (id: number): ItemType => {
+    if (this.props.currentMeshFile != null) {
+      return {
+        key: "loadMeshesOfGroup",
+        label: (
+          <>
+            <CloudDownloadOutlined />
+            Load all meshes
+          </>
+        ),
+        children: [
+          {
+            key: "loadAdHoc",
+            label: (
+              <div
+                onClick={() => {
+                  if (this.props.visibleSegmentationLayer == null) {
+                    return;
+                  }
+                  this.handleLoadMeshesAdHoc(id);
+                  this.getToastForMissingLocations(id);
+                  this.handleSegmentDropdownMenuVisibility(id, false);
+                }}
+              >
+                ad hoc
+              </div>
+            ),
+          },
+          {
+            key: "loadByFile",
+            label: (
+              <div
+                onClick={() => {
+                  if (this.props.visibleSegmentationLayer == null) {
+                    return;
+                  }
+                  this.handleLoadMeshesFromFile(id);
+                  this.getToastForMissingLocations(id);
+                  this.handleSegmentDropdownMenuVisibility(id, false);
+                }}
+              >
+                precomputed
+              </div>
+            ),
+          },
+        ],
+      };
+    } else {
+      return {
+        key: "loadOnlyAcHoc",
+        label: (
+          <div
+            onClick={() => {
+              if (this.props.visibleSegmentationLayer == null) {
+                return;
+              }
+              this.handleLoadMeshesAdHoc(id);
+              this.getToastForMissingLocations(id);
+              this.handleSegmentDropdownMenuVisibility(id, false);
+            }}
+          >
+            <CloudDownloadOutlined />
+            Load meshes for segments (ad-hoc)
+          </div>
+        ),
+      };
+    }
+  };
+
+  getReloadMenuItem = (groupId: number): ItemType => {
+    return this.state != null && this.doesGroupHaveAnyMeshes(groupId)
+      ? {
+          key: "reloadMeshes",
+          label: (
+            <div
+              onClick={() => {
+                if (this.props.visibleSegmentationLayer == null) {
+                  return;
+                }
```

Comment:
I suspect that this guard (here and probably in all the other onclick handlers) can be removed, because visibleSegmentationLayer is not used in the onclick handler per se. instead, the functions which are called (e.g., handleRefreshMeshes) do the same check again.